src/app/views/discover/games/list/list.ts

```diff
@@ -106,98 +132,148 @@ export default class RouteDiscoverGamesList extends BaseRouteComponent {
 			});
 		}
 
-		if (this.tag) {
-			switch (this.section) {
-				case 'featured':
-					return this.$gettextInterpolate('Featured %{tag} games', { tag: this.tag });
-
-				case 'best':
-					return this.$gettextInterpolate('Best %{tag} games (top-rated)', {
-						tag: this.tag,
-					});
+		let onlyFree = this.filtering && this.filtering.getFilter('price') === 'free';
 
-				case 'new':
-					return this.$gettextInterpolate('New %{tag} games', { tag: this.tag });
+		const title = this.$gettextInterpolate('%{ listOf } %{ free } %{ gamesType }', {
+			listOf: this.displayListOf,
+			free: onlyFree ? this.$gettext('free') : '',
+			gamesType: this.displayGamesType,
+		});
 
-				default:
-					return this.$gettextInterpolate('Browsing %{tag} games', { tag: this.tag });
-			}
-		}
+		// Remove duplicate whitespaces (formed when a replacement ends up being an empty string)
+		return title.replace(/\s+/g, ' ');
+	}
 
+	get displayListOf() {
 		switch (this.section) {
 			case 'featured':
-				return this.$gettext('Featured games');
-
+				return this.$gettext('Featured');
 			case 'best':
-				return this.$gettext('Best games (top-rated)');
-
+				return this.$gettext('Best');
 			case 'new':
-				return this.$gettext('New games');
+				return this.$gettext('Newest');
+			default:
+				return this.$gettext('Browse');
+		}
+	}
+
+	get displayGamesType() {
+		if (!this.tag) {
+			return this.$gettext('games');
+		}
 
+		switch (this.tag) {
+			case 'horror':
+				return this.$gettext('scary horror games');
+			case 'fangame':
+				return this.$gettext('fan games');
+			case 'fnaf':
+				return this.$gettext(`Five Nights at Freddy's games`);
+			case 'scifi':
+				return this.$gettext('science fiction games');
+			case 'retro':
+				return this.$gettext('nostalgic and retro games');
+			case 'pointnclick':
+				return this.$gettext('point and click games');
+			case 'altgame':
+				return this.$gettext('alt games');
+			case 'roguelike':
+				return this.$gettext('rogue like games');
```

Comment:
Roquelike is one word: https://en.wikipedia.org/wiki/Roguelike

Reply:
dang